syncapi/synctypes/clientevent.go

```diff
@@ -44,6 +66,9 @@ type ClientEvent struct {
 	Type           string         `json:"type"`
 	Unsigned       spec.RawJSON   `json:"unsigned,omitempty"`
 	Redacts        string         `json:"redacts,omitempty"`
+
+	// Federation Event Fields. Only sent to clients when `event_format` == `federation`.
+	ClientFederationEvent
```

Comment:
I could rename it maybe to be `ClientFederationFields` or something. That would make more sense.
Unfortunately we use the terminology `ClientEvent` all over the place so there would be a lot of changes to make if we wanted to flip it. 

In high level terms, I think it makes sense for us to be using `ClientEvent` everywhere since it's what we are sending to the client. Whether or not it is formatted as a federation event then gets layered on top of the `ClientEvent`.